Assets/Scripts/UI/UI.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using PlayerInput;
+
+public class UI : MonoBehaviour
+{
+    public Image AbilityCoolDown;
+    public Image Ability;
+    public Image PreviousAbility;
+    public Image Grapple;
+    public Image GrappleCooldown;
+    public Sprite GreenAbilitySprite;
+    public Sprite RedAbilitySprite;
+    public Sprite YellowAbilitySprite;
+    public Sprite BlueAbilitySprite;
+    public Sprite GreenSprite;
+    public Sprite RedSprite;
+    public Sprite YellowSprite;
+    public Sprite BlueSprite;
+    private InputPlayer inputPlayer;
+    private ColorState playerColor;
+    private PlayerMovement playerMoevement;
+
+    public GameColor storedColor;
+
+    public float coolDown = 5;
+
+    void Awake()
+    {
+        playerColor = gameObject.transform.parent.transform.parent.GetComponentInChildren<ColorState>();
+        playerMoevement = gameObject.transform.parent.transform.parent.GetComponentInChildren<PlayerMovement>();
+    }
+
+        // Use this for initialization
+        void Start ()
+    {
+        Green();
+    }
+	
+
+	void Update ()
+    {
+        if (InputManager.GetButtonDown(PlayerButton.Swap, inputPlayer))
+        {
+            GameColor temp = storedColor;
+            storedColor = playerColor.currentColor;
+            playerColor.currentColor = temp;
```

Comment:
Having an extra check for the button press will definitely cause issues
add an update event to the colorswap event list which includes a check to the---
oh I just realized you removed quickswap entirely and copied things over

Please remove the quick swap logic from this class and add it back to that class
Since quickswap is a feature available on some levels and not others we need to let design disable or enable that script 

Make sure that this script works regardless of if quickswap is enabled or not (if quickswap is inactive then dont display the backup color)